internal/repos/store_test.go

```diff
@@ -308,7 +296,128 @@ func TestStoreEnqueueSingleSyncJob(t *testing.T) {
+				t.Fatal(err)
+			}
+		},
+		"EnqueueSyncJobs": func(t *testing.T, ctx context.Context, store repos.Store, _ *types.ExternalService) {
+			t.Helper()
+			if err := store.EnqueueSyncJobs(ctx, false); err != nil {
+				t.Fatal(err)
+			}
+		},
+	}
+
+	for name, enqueuingFunc := range enqueuingFuncs {
+		t.Run(name, func(t *testing.T) {
+			t.Cleanup(func() {
+				q := sqlf.Sprintf("DELETE FROM external_service_sync_jobs;DELETE FROM external_services")
+				if _, err := store.Handle().ExecContext(ctx, q.Query(sqlf.PostgresBindVar), q.Args()...); err != nil {
+					t.Fatal(err)
+				}
+			})
+
+			service := types.ExternalService{
+				Kind:        extsvc.KindGitHub,
+				DisplayName: "Github - Test",
+				Config:      extsvc.NewUnencryptedConfig(`{"url": "https://github.com", "repositoryQuery": ["none"], "token": "abc"}`),
+				CreatedAt:   now,
+				UpdatedAt:   now,
+			}
+
+			// Create a new external service
+			err := database.ExternalServicesWith(logger, store).Create(ctx, confGet, &service)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// Sanity check: can create sync jobs when service is not locked/being deleted
+			assertSyncJobCount(t, store, 0)
+			enqueuingFunc(t, ctx, store, &service)
+			assertSyncJobCount(t, store, 1)
+
+			// Mark jobs as completed so we can enqueue new one (see test above)
+			q := sqlf.Sprintf("UPDATE external_service_sync_jobs SET state='completed'")
+			if _, err = store.Handle().ExecContext(ctx, q.Query(sqlf.PostgresBindVar), q.Args()...); err != nil {
+				t.Fatal(err)
+			}
+
+			// If the external service is selected with FOR UPDATE in another
+			// transaction it shouldn't enqueue a job
+			//
+			// Open a transaction and lock external service in there
+			tx, err := store.Transact(ctx)
+			if err != nil {
+				t.Fatalf("Transact error: %s", err)
+			}
+			q = sqlf.Sprintf("SELECT id FROM external_services WHERE id = %d FOR UPDATE", service.ID)
+			if _, err = tx.Handle().ExecContext(ctx, q.Query(sqlf.PostgresBindVar), q.Args()...); err != nil {
+				t.Fatal(err)
+			}
+
+			// In the background delete the external service and commit
+			done := make(chan struct{})
+			go func() {
+				time.Sleep(500 * time.Millisecond)
+
+				q := sqlf.Sprintf("UPDATE external_services SET deleted_at = now() WHERE id = %d", service.ID)
+				if _, err = tx.Handle().ExecContext(ctx, q.Query(sqlf.PostgresBindVar), q.Args()...); err != nil {
+					logger.Error("deleting external service failed", log.Error(err))
+				}
+				if err = tx.Done(err); err != nil {
+					logger.Error("commit transaction failed", log.Error(err))
+				}
+				close(done)
+			}()
+
+			// This blocks until transaction is commited and should NOT have enqueued a
+			// job because external service is deleted
+			enqueuingFunc(t, ctx, store, &service)
+			assertSyncJobCount(t, store, 1)
+
+			// Sanity check: wait for transaction to commit
+			select {
+			case <-time.After(5 * time.Second):
+				t.Fatalf("timeout!")
```

Comment:
adding a bit more verbosity here
```suggestion
				t.Fatalf("background goroutine deleting external service timed out!")
```